### ISSUE TITLE
[BUG]:Grant-PnPAzureADAppSitePermission

### ISSUE THREAD

**Issue by cbtboss** [pnp/powershell#2980]
### Reporting an Issue or Missing Feature
Using Grant-PnPAzureADAppSitePermission to grant write access to sites.selected errors out while using 2.1.1 stating it is an invalid request. When doing the exact same command with 1.11 it works (I have a vm with the new version and one with the old). 

### Expected behavior 
A new azureAd app site permission should be created. 

### Actual behavior 
No new AzAD appsite permission is created and a generic invalid request error is returned. 

### Steps to reproduce behavior
```
#replaced my actual domain/host/site with contoso
$siteUrl = 'https://contoso.sharepoint.com/sites/contososite'
connect-pnponline -interactive -url $siteUrl -tenant contoso.onmicrosoft.com
Grant-PnPAzureADAppSitePermission -Permissions "Write" -Site $siteUrl -AppId $appId -DisplayName "SitesResourceSpecific"
```

### What is the version of the Cmdlet module you are running?
2.1.1 is the one that is failing. It works for me on 1.11 exact same cmdlets. 
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by OS-Lee** [comment]
Same issue.

**Comment by gautamdsheth** [comment]
Have fixed this issue, it will be available in tomorrow's nightly builds.

Please let us know if the issue still persists.

**Comment by erkanch** [comment]
Hi @gautamdsheth Do you have an ETA on when this fix will be released? Thanks!

**Comment by Bartdejonge06** [comment]
Had the same issue, Prerelease fixed it! Thanks @gautamdsheth!

**Comment by amardeep2006** [comment]
> After upgrading to the today's nightly, the internal error is gone - however I now get:
> 
> > Forbidden (403): Access denied
> 
> Since I also get this when calling Get-PnPAzureADAppSitePermission, I assume that this points to a different issue than the one reported by OP: While the documentation states that Sites.FullControl.All is the required permission (which is granted), this does not seem to be the case - at least when using an own Azure AD app and connecting with Client ID and PFX.

This can happen if you are not site admin. Replying in case someone does Internet Search for 403 error with Get-PnPAzureADAppSitePermission this and lands here. I faced same issue as I had FullControl over site but was not site admin. Menu Settings --> Site Permissions --> Advanced Permissions Settings --> Site Collection Administrators

**Comment by ComputerHabit2** [comment]
I just made a new app with full rights.  I can login and see all my sites as admin.

Grant-PnPAzureADAppSitePermission -AppId "$($appid)" -Site $siteUrl -DisplayName "test" -Permissions "write"

Fails
Grant-PnPAzureADAppSitePermission: {"error":{"code":"accessDenied","message":"Access denied","innerError":{"date":"2024-09-27T18:38:10","request-id":"xxxx112fb5f","client-request-id":"xxxxx3112fb5f"}}}

**Comment by cbtboss** [comment]
@ComputerHabit2 that is unrelated to the original issue that I submitted here. That error MOST likely refers to that the user you are running as needs to be an Owner of the sharepoint site. This is true regardless of if you run it as a Global Admin.

**Comment by ComputerHabit2** [comment]
> @ComputerHabit2 that is unrelated to the original issue that I submitted here. That error MOST likely refers to that the user you are running as needs to be an Owner of the sharepoint site. This is true regardless of if you run it as a Global Admin.

I agree.  This was me getting stuck with switches being disabled and having to scramble for understanding of the changes.
